### DISCUSSION TITLE
Need help with a somewhat complex search

### DISCUSSION THREAD

**Discussion by JulieVelghe** [craftcms/cms#13827] [Q&A] [answered]
Hi all, I need some help with some searching logic.
I want to search for a term in a searchbox, across multiple domains and multiple sections. So far, that works.
Now, I want to restrict the results from one of these sections.
Lets say, I search in the sections 'events', 'partners', 'equipment'. I want to restrict the events, where a field 'relatedThemes' (which is an Entry), has the slug 'hiking'.

I could do a check on this 'relatedThemes' field after I've run the query, but that original query gets used for pagination. So there would be a mismatch in counts between the two.

So ideally, I would want to add this restriction on to my query.

I was thinking about using relatedTo, something to this effect:

```
$relTheme = Entry::find()
	->slug('hiking')
	->one();
$entries = Entry::find()
            ->siteId([1,2])
            ->search($query)
            ->orderBy('score');

$entries->section(['events','partners','equipment']);

$entries->relatedTo($relTheme);

```

However, I'm not sure as to how this would behave, with regards to the other sections. I have a feeling that this setup would only give me events related to 'hiking', and ommit 'partners' and 'equipment' alltogether.

Any tips or insights in how I could set this up would be greatly appreciated!

**Comment by michtio** [comment]
@JulieVelghe you could simply let your query end and return a collection, then you can use collection methods to filter your data further.

**Reply by JulieVelghe** [reply]
I could, but then I would lose the ability to paginate the results with the {% paginate %} tag, as the amount of filtered entries will be smaller that the results from the original query.

**Comment by brandonkelly** [comment, marked as answer]
One way to do it is you could start by getting all the entry IDs you _don’t_ want, and then telling your main query to exclude them.

```php
$hikingId = Entry::find()
    ->section('themes') // change this as needed
    ->slug('hiking')
    ->ids();

$hikingThemedEntryIds = Entry::find()
    ->section(['events', 'partners', 'equipment'])
    ->relatedTo([
        'targetElement' => $hikingId,
        'field' => 'relatedThemes'
    ])
    ->ids();

$entries = Entry::find()
    ->section(['events', 'partners', 'equipment'])
    ->id(array_merge(['not'], $hikingThemedEntryIds))
    ->siteId([1, 2])
    ->search($query)
    ->orderBy('score');
```
